rewrite-pattern.js

```diff
@@ -299,19 +301,23 @@ const config = {
 	'ignoreCase': false,
 	'unicode': false,
 	'dotAll': false,
-	'useUnicodeFlag': false
+	'useUnicodeFlag': false,
+	'unicodePropertyEscape': false,
+	'namedGroup': false
 };
 const rewritePattern = (pattern, flags, options) => {
+	config.unicode = flags && flags.includes('u');
 	const regjsparserFeatures = {
-		'unicodePropertyEscape': options && options.unicodePropertyEscape,
-		'namedGroups': options && options.namedGroup,
+		'unicodePropertyEscape': config.unicode,
```

Comment:
Didn't you just delete the line that defines `config.unicode`?

Reply:
I don't think we can delete `config.unicode` as the `bmpOnly` regenerate option also relies on `config.unicode`.

Reply:
Never mind, I missed that you just moved it to the top.